src/mlpack/core/data/load.cpp

```diff
@@ -50,7 +50,44 @@ template bool Load<double>(const std::string&,
                            const bool,
                            const bool,
                            const arma::file_type);
+*/
+//---------------------------------------------------------
+template bool Load<int>(const std::string&,
+                        arma::Mat<int>&,
+                        const bool,
+                        const bool,
+                        const file_type);
+
+template bool Load<unsigned int>(const std::string&,
+                                 arma::Mat<unsigned int>&,
+                                 const bool,
+                                 const bool,
+                                 const file_type);
+
+template bool Load<unsigned long>(const std::string&,
+                                  arma::Mat<unsigned long>&,
+                                  const bool,
+                                  const bool,
+                                  const file_type);
 
+template bool Load<unsigned long long>(const std::string&,
+                                       arma::Mat<unsigned long long>&,
+                                       const bool,
+                                       const bool,
+                                       const file_type);
+
+template bool Load<float>(const std::string&,
+                          arma::Mat<float>&,
+                          const bool,
+                          const bool,
+                          const file_type);
+
+template bool Load<double>(const std::string&,
+                           arma::Mat<double>&,
+                           const bool,
+                           const bool,
+                           const file_type);
+// --------------------------------------------------------------------
```

Comment:
I do not think we need all of these overloads, even we can delete entirely this file according to the modification we are doing now. @rcurtin let me know if I am right here

Reply:
Agreed---we should be able to remove the explicit instantiations in this file like these. :+1: